Docs/BUILDING.md

```diff
@@ -45,13 +45,31 @@ At this point you have checked out the source code for both the Firefox for iOS
 
 Everything after this point is done from within Xcode.
 
+Pointing to Local Rust Components (Application Services)
+--------------------
+
+Firefox for iOS depends internally on some of the [shared Rust components](https://github.com/mozilla/application-services). Sometimes, you may want to also point to your local Rust components when building locally. You can do so by:
+
+1. First ensure you can [build application-services](https://github.com/mozilla/application-services/blob/master/docs/building.md) locally.
+2. Next, `carthage build --no-skip-current --platform iOS --verbose --configuration Debug --cache-builds`.
+3. Now back in firefox-ios, after `carthage bootstrap`, replace the application-services library with a symlink:
+
+  ```
+  rm -rf Carthage/Build/iOS/MozillaAppServices.framework
+  ln -s ~/REPLACE_WITH_PATH_TO_YOUR_LOCAL_CHECKOUT/application-services/Carthage/Build/iOS/Static/MozillaAppServices.framework Carthage/Build/iOS
+  ```
+
+4. Build firefox-ios.
+
+Every time you make a change to application-services, re-run the carthage build command shown in step #2, maybe clear fx iOS xcode cache with Cmd+k and you're good to go.
```

Comment:
nit: `maybe clear fx iOS xcode cache with Cmd+k and you're good to go.` -> I have never had to do this, is this a problem you ran into?

Reply:
yeah, actually I did not have to do it. This came as a suggestion from @eoger who provided me with these details. I can just remove it for now since you're right; it's more generic than specific to this process.